chia/wallet/nft_wallet/nft_puzzles.py

```diff
@@ -207,6 +209,53 @@ def get_percentage_from_puzzle(puzzle: Program) -> Optional[uint64]:
     return None
 
 
+def get_nft_info_from_puzzle(puzzle: Program, nft_coin: Coin) -> NFTInfo:
+    """
+    Extract NFT info from a full puzzle
+    :param puzzle: NFT full puzzle
+    :param nft_coin: NFT coin
+    :return: NFTInfo
+    """
+    # TODO Update this method after the NFT code finalized
+    curried_args = match_nft_puzzle(puzzle)[1]
+    (
+        _,
+        singleton_struct,
+        current_owner_did,
+        nft_transfer_program_hash,
+        transfer_program_curry_params,
+        metadata,
+    ) = curried_args
+    (
+        royalty_address,
+        trade_price_percentage,
+        settlement_mod_hash,
+        cat_mod_hash,
+    ) = transfer_program_curry_params.as_iter()
+    for kv_pair in metadata.as_iter():
+        if kv_pair.first().as_atom() == b"u":
+            data_uris = []
```

Comment:
I'd move this line right above the for loop to ensure that it's both declared and initialized to an empty list. If the "u" key isn't found in the metadata, NFTInfo would be constructed with data_uris == None